pandas/tests/groupby/test_function.py

```diff
@@ -179,7 +179,13 @@ def test_arg_passthru():
         tm.assert_index_equal(result.columns, expected_columns_numeric)
 
         result = f(numeric_only=False)
-        tm.assert_frame_equal(result.reindex_like(expected), expected)
+
+	# TODO: median isn't implemented for DTI but was working blockwise before?
```

Comment:
Surprised by this as well. I guess the by block iteration would bypass some of the checks that were put into a DTA, so while this doesn't work on master:

```python-traceback
>>> df = pd.DataFrame([pd.Timestamp("2000-01-01"), pd.Timestamp("2000-01-01")])
>>> df[0].median()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-18-c894fad0307b> in <module>
----> 1 df[0].median()

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/generic.py in stat_func(self, axis, skipna, level, numeric_only, **kwargs)
  11616             return self._agg_by_level(name, axis=axis, level=level, skipna=skipna)
  11617         return self._reduce(
> 11618             f, name, axis=axis, skipna=skipna, numeric_only=numeric_only
  11619         )
  11620

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/series.py in _reduce(self, op, name, axis, skipna, numeric_only, filter_type, **kwds)
   4097             numeric_only=numeric_only,
   4098             filter_type=filter_type,
-> 4099             **kwds
   4100         )
   4101

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/base.py in _reduce(self, op, name, axis, skipna, numeric_only, filter_type, **kwds)
   1218             raise TypeError(
   1219                 "{klass} cannot perform the operation {op}".format(
-> 1220                     klass=self.__class__.__name__, op=name
   1221                 )
   1222             )

TypeError: DatetimeIndex cannot perform the operation median
```

This somehow did

```python
>>> df.groupby([1, 1]).median(numeric_only=False)
           0
1 2000-01-01
```

Whether or not that is intentional I'm not sure so cc @jbrockmendel

Reply:
Not sure off the top of my head what is going on here, but I can say that DTA/DTI _should_ have `median`.  #28165 implements reductions for TDA, waiting to get that merged before doing DTA and other reductions.

Reply:
Yea I just saw the update to that come through. I'll try to give it another look review wise tomorrow. Might be the precursor we need for this

Reply:
> Might be the precursor we need for this

Any thoughts on how this interacts with the ongoing catch-less project?  I'm pretty sure the block-wise method this rips out is involved in a lot of that catching.

Reply:
Tough to say but I think it can only help. I think generally a lot of the blanket Exception catching and state management in GroupBy might make it one of the harder parts of the code base to digest, so limiting the ways in which we construct results here should make things easier in the future (lot more to be done)